src/EditorFeatures/Core/Implementation/InlineRename/CommandHandlers/RenameCommandHandler_SaveHandler.cs

```diff
@@ -17,7 +17,17 @@ public void ExecuteCommand(SaveCommandArgs args, Action nextHandler)
         {
             if (_renameService.ActiveSession != null)
             {
-                _renameService.ActiveSession.Commit();
+                if (!_renameService.ActiveSession.TryCommit())
+                {
+                    // If the user cancels the Preview Changes window during a save, then we would
+                    // ideally leave the session active. However, if this save command was issued
+                    // in preparation for a debugging session then we're better off cancelling the
+                    // session now because once the debugging session starts we will be forced to
+                    // completely abandon the session instead of cleanly rolling back. See bug
+                    // #1176673 for more details.
+                    _renameService.ActiveSession.Cancel();
```

Comment:
It seems we really should be abandoning the save entirely, right? Otherwise I'm worried that we create a very strange experience; sometimes hitting cancel in the preview will just take you back to the rename session, but sometimes it throws away the rename entirely.